doc/dev/development.md

```diff
@@ -1,336 +1,12 @@
 # Development Guide
```

Comment:
Let's remove this file completely and just point the README.md (along with any other references) directly to the istio/istio location.